java/src/org/openqa/selenium/remote/RemoteWebDriver.java

```diff
@@ -1248,12 +1249,9 @@ public String getId() {
 
     @Override
     public void addCredential(Credential credential) {
-      execute(
-          DriverCommand.ADD_CREDENTIAL,
-          Stream.concat(
-                  credential.toMap().entrySet().stream(),
-                  Stream.of(Map.entry("authenticatorId", id)))
-              .collect(Collectors.toUnmodifiableMap(Map.Entry::getKey, Map.Entry::getValue)));
+      Map<String, Object> map = new HashMap<>(credential.toMap());
+      map.put("authenticatorId", id);
+      execute(DriverCommand.ADD_CREDENTIAL, map);
```

Comment:
do we want `Map.copyOf(map)` here to ensure immutability? Not certain it matters.